lib/util/http/github.ts

```diff
@@ -301,20 +302,40 @@ export class GithubHttp extends Http<GithubHttpOptions, GithubHttpOptions> {
     const result: T[] = [];
 
     const { paginate = true } = options;
-    let count = options.count || 100;
+
+    const initialCount = options.count || 100;
+    let count = initialCount;
     let limit = options.limit || 1000;
     let cursor: string = null;
 
     let isIterating = true;
     while (isIterating) {
+      const actualCount = Math.min(count, limit);
       const res = await this.requestGraphql<GithubGraphqlRepoData<T>>(query, {
         ...options,
-        count: Math.min(count, limit),
+        count: actualCount,
         cursor,
         paginate,
       });
       const fieldData = res?.data?.repository?.[fieldName];
       if (fieldData) {
+        // See: https://github.com/renovatebot/renovate/issues/12703
+        // istanbul ignore if
+        if (
+          count < initialCount &&
+          count === actualCount &&
+          baseUrl === githubBaseUrl
+        ) {
+          const { owner, name } = options?.variables || {};
+          const repo = owner && name ? `${owner}/${name}` : null;
+          if (repo) {
+            logger.debug(
+              { repo, count },
```

Comment:
I'm pretty sure the repo should already be logged due to existing metadata? Do we ever use graphql to query a repo other than config.repository?

Reply:
Don't think so. Fixed.